Tools/autotest/common.py

```diff
@@ -717,11 +717,26 @@ def do_get_autopilot_capabilities(self):
                 return
         raise AutoTestTimeoutException()
 
-    def do_set_mode_via_command_long(self):
+    def get_mode_from_mode_mapping(self, mode):
+        """Validate and return the mode number from a string or int."""
+        mode_map = self.mav.mode_mapping()
+        if mode_map is None:
+            raise ErrorException()
+        if isinstance(mode, str):
+            if mode in mode_map:
+                return mode_map.get(mode)
+        if mode in mode_map.values():
+            return mode
+        self.progress("Unknown mode '%s'" % mode)
+        self.progress("Available modes '%s'" % mode_map)
+        raise ErrorException()
+
+    def do_set_mode_via_command_long(self, mode, timeout=60):
```

Comment:
That's a long timeout... My understanding is that you want to mimic the wait_mode method, but in that case I think the recv_math call is the one that should have the long timeout.